user_interface.py

```diff
@@ -0,0 +1,70 @@
+import os.path
+import phonebook_utils as func
+
+
+def menu():
+    while True:
+        print('\nВас приветствует Телефонный справочник.\n'
+              '\nВыберете пункт меню:\n'
+              '\n1 - Показать все записи в телефонном справочнике.\n'
+              '2 - Найти запись в справочнике по имени.\n'
+              '3 - Найти запись в справочнике по фамилии.\n'
+              '4 - Найти запись в справочнике по дате рождения.\n'
+              '5 - Найти запись в справочнике по месту работы.\n'
+              '6 - Поиск записи в справочнике по номеру телефона.\n'
+              '7 - Добавить новую запись.\n'
+              '8 - Закрыть телефонный справочник.\n')
+        number_menu = input('Введите порядковый номер пункта меню: ')
+
+        if number_menu.isdigit() and int(number_menu) in range(1, 9):
+            result_print = []
+
+            if int(number_menu) == 1:
+                if os.path.exists('phonebook.csv'):
+                    result_print = func.read_phonebook()
+                else:
+                    count = int(input('Задайте необходимое количество строк в '
+                                      'телефонном справочнике: '))
+                    result_print = func.create_phonebook(count)
+
+            elif int(number_menu) == 2:
+                search_name = input('Введите имя: ')
+                result_print = func.search_phonebook(name=search_name)
+
+            elif int(number_menu) == 3:
+                search_surname = input('Введите фамилию: ')
+                result_print = func.search_phonebook(surname=search_surname)
+
+            elif int(number_menu) == 4:
+                search_birthday = input('Введите дату рождения: ')
+                result_print = func.search_phonebook(birthday=search_birthday)
+
+            elif int(number_menu) == 5:
+                search_work_place = input('Введите место работы: ')
+                result_print = func.search_phonebook(
+                    work_place=search_work_place)
+
+            elif int(number_menu) == 6:
+                search_number = input('Введите номер телефона: ')
+                result_print = func.search_phonebook(
+                    phone_number=search_number)
```

Comment:
А почему бы вам не создать переменную, что то типа словаря, где каждому варианту ответа будет назначено значение того или иного параметра для поиска. Тогда можно код сократить до 
```suggestion
            elif number_menu in ('2', '3', '4', '5', '6'):
               some_dict = {
                    '2': ['name', input('Введите имя: ')],
                    '3': ['surname', input('Введите фамилию: ')],
                    '4': ['birthday', input('Введите дату рождения: ')],
                    '5': ['work_place', input('Введите место работы: ')],
                    '6': ['phone_number', input('Введите номер телефона: ')],
                }
                result_print = func.search_phonebook(some_dict[number_menu][0]=some_dict[number_menu][1]))
```
Как-то так

Reply:
отличная идея, я сама не дошла до этого) возьму на будущее)